src/transaction/validation.rs

```diff
@@ -29,6 +90,8 @@ impl Input {
                 Err(ValidationError::InputCoinWitnessIndexBounds { index })
             }
 
+            Self::Coin { .. } => Ok(()),
```

Comment:
Its separated because of the `TODO` there: https://github.com/FuelLabs/fuel-tx/blob/d37cb3b0cdc02ed120f7ccbe5648fb8c8205a228/src/transaction/validation.rs#L47-L49